crud/helpers.go

```diff
@@ -136,14 +138,27 @@ func LoadBalancerResourceGet(s BaseCrud, workReq *baremetal.WorkRequest) (id str
 	return id, false, nil
 }
 
+func exponentialBackoffSleep(retryNum uint) {
+	secondsToSleep := 1 << retryNum
+	log.Printf("[DEBUG] Got a retriable error. Waiting %d seconds and trying again...", secondsToSleep)
+	time.Sleep(time.Duration(secondsToSleep) * time.Second)
+}
+
+func isRetriableError(error string) bool {
+	//all errors are retriable except 401, 403, and 404
+	match, _ := regexp.MatchString("status\\s*:\\s*40[134]", strings.ToLower(error))
+	return !match
+}
+
 func CreateResource(d *schema.ResourceData, sync ResourceCreator) (e error) {
+	return createResourceWithRetry(d, sync, 1)
+}
+
+func createResourceWithRetry(d *schema.ResourceData, sync ResourceCreator, retryNum uint) (e error) {
 	if e = sync.Create(); e != nil {
-		// Check for conflicts and retry
-		// This happens with concurrent volume attachments, etc
-		if strings.Contains(strings.ToLower(e.Error()), "try again later") {
-			log.Println("[DEBUG] Resource creation conflicts with other resources. Waiting 10 seconds and trying again...")
-			time.Sleep(10 * time.Second)
-			e = CreateResource(d, sync)
+		if isRetriableError(e.Error()) && retryNum <= MaxRetries {
```

Comment:
Why not pass the error itself and test the code field directly rather than stringifying and regexing on the string.